### ISSUE TITLE
Need Ditto to filter certain text phrases.

### ISSUE THREAD

**Issue by Uzukii** [sabrogden/Ditto#464]
I would like for Ditto to ignore HTML code. For example
{outlinecolor=#00aedb}I wish...{/outlinecolor}

Is it possible to make it only copy and send the "I wish..." part and ignore anything within brackets? Not just the specific text in the example, but any text by using wildcards. {outlinecolor=#******}

**Comment by bradacjan** [comment]
Hello, a long-time Ditto user here. I do not know any way to do what you are describing. And I am looking for a similar feature (more below).

I suspect there may be a way to define your own Supported Type that would have custom behaviour but I have no experience with that. Or, perhaps, seeing how and where Special Paste (e.g., CamelCase) is defined and add some more generic option. But that would probably require a pull request.

I am currently looking for a way to remove gclid from URL that I copy. _Since it usually represents about 80% of the link's length and I manually delete it every time (because I find it very annoying when saving or sharing links), I am looking for a way to always remove that using Ditto._ I assume that both our needs would be solved by a simillar functionality – **some regex replacement performed when copying or pasting** (regex or some more complex logic).

I would appreciate if some more knowledgeable user could point us towards ways to implement that.

**Comment by manasan3010** [comment]
@Uzukii @bradacjan you can achieve this with custom ChaiScript. Ditto suports onCopy and onPaste custom scripts.

More info here: https://github.com/sabrogden/Ditto/wiki/Scripting

**Comment by Uzukii** [comment]
@manasan3010 Thanks for that. Even though it was staring me in the face the whole time, I didn't know how to use it. My situation has changed a bit though. I no longer need a filter function but I could definitely use a replacer function. Where "Word / Phrase A" gets replaced with "Word / Phrase B" on copy. But the replacement only happens if the variable matches it exactly. Such as being all lower case letters and not part of another word. Can Ditto/ChaiScript do that?

Basically, I want Notepad++'s replace feature that has checkboxes for "match whole word only" and "match case" to be done on the fly.
